cosmos/scheduler/cmd/scheduler.go

```diff
@@ -138,19 +145,18 @@ func (s *Scheduler) watchOverlayUpdates() (chan interface{}, error) {
 		return nil, err
 	}
 	go func() {
-		for {
-			if cur.Next(ctx) {
-				ch <- "overlay update"
-			}
-			if err := cur.Err(); err != nil {
-				log.Println(err)
-			}
-			err := s.configureIngestionSecret()
-			if err != nil {
+		defer cur.Close(ctx)
```

Comment:
I'm not sure if the race detector would be happy about accessing `cur` here in a different goroutine than where it's created and without locking. I suggest instanciating as many of the concurrency-unsafe objects as possible in the goroutine (the channel is fine though). It would also fix the quirk of deferring a close on something opened in another scope.

Reply:
Yeah that makes sense. There's definitely lots more that need to be fixed/updated. It's clear to me that I need to spend more time on these changes so I think I'll drop this commit from this PR and open a separate PR with your suggestions/considerations.